server/commands/list_objects.go

```diff
@@ -0,0 +1,279 @@
+package commands
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/tuple"
+	serverErrors "github.com/openfga/openfga/server/errors"
+	"github.com/openfga/openfga/storage"
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/unit"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	MaximumConcurrentChecks = 100 // todo(jon-whit): make this configurable, but for now limit to 100 concurrent checks
+)
+
+type ListObjectsQuery struct {
+	Datastore             storage.OpenFGADatastore
+	Logger                logger.Logger
+	Tracer                trace.Tracer
+	Meter                 metric.Meter
+	ListObjectsDeadline   time.Duration
+	ListObjectsMaxResults uint32
+	ResolveNodeLimit      uint32
+}
+
+// Execute the ListObjectsQuery, returning a list of object IDs
+func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjectsRequest) (*openfgapb.ListObjectsResponse, error) {
+	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
+	if err != nil {
+		return nil, err
+	}
+
+	allResults, _ := q.Meter.SyncInt64().Counter(
+		"openfga.listobjects.allresults",
+		instrument.WithDescription("Number of object IDs returned"),
+		instrument.WithUnit(unit.Dimensionless),
+	)
+	partialResults, _ := q.Meter.SyncInt64().Counter(
+		"openfga.listobjects.partialresults",
+		instrument.WithDescription("Number of object IDs returned after the API timed out"),
```

Comment:
I would think we would count the total requests which were partial, not the total results.

Reply:
Agreed. If we're going to track a metric like this, let's count the requests that returned because of the `listObjects-deadline` being met. Though, what use do we have for the metric at this time? If we don't have a strong reason to introduce a metric to track some defined SLO, then I'd say we don't introduce a new metric at all. We can always add it later if we have a need for it then.

Reply:
> what use do we have for the metric at this time

satisfy my own curiosity of how many requests terminate on time 😜

Reply:
> satisfy my own curiosity
That's not a very strong reason to track a metric though 😄 . It has no relationship with any SLOs we publicly declare as being important for a developer to monitor OpenFGA health/performance.

If we have an initiative to track this because it's relevant to a broader monitoring objective, then let's do it. Otherwise let's leave it out until it has a stronger purpose.

Reply:
how are we going to find out if our deadline timeout is too low if we don't emit this metric? 

also metrics aren't always about SLOs, if i'm oncall i want to know this kind of things